src/Rinds/RindsSpecsDataSourcePoller.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Handles polling and storage of RINDS specs
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\Rinds;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * RINDS specs data source poller class.
+ * This handles polling RINDS specs from JSON endpoints, and
+ * stores the specs in to the database as an option.
+ */
+class RindsSpecsDataSourcePoller {
+	const DATA_SOURCES = array(
+		'http://two.wordpress.test/rinds-specs.json',
+	);
+
+	/**
+	 * Polls the data sources for specs, persists those specs, and delete any
+	 * specs that no longer exist in the data sources.
+	 */
+	public static function poll_data_sources() {
+		$specs = array();
+
+		// Note that this merges the specs from the data sources based on the slug - last one wins.
+		foreach ( self::DATA_SOURCES as $url ) {
+			self::read_data_source( $url, $specs );
+		}
+
+		// Persist the specs as an option.
+		$existing_option = get_option( RindsEngine::SPECS_OPTION_NAME );
+
+		if ( false === $existing_option ) {
+			add_option( RindsEngine::SPECS_OPTION_NAME, $specs );
+		} else {
+			update_option( RindsEngine::SPECS_OPTION_NAME, $specs );
```

Comment:
`update_option()` will add the option if it does not exist, so we can forego this `if/else` logic with a single line to update the option.

Reply:
Nice - I think I copied that from somewhere else